modules/installation-creating-gcp-iam-shared-vpc.adoc

```diff
@@ -0,0 +1,123 @@
++
+----
+$ export MASTER_NAT_IP=`gcloud compute addresses describe ${INFRA_ID}-master-nat-ip --region ${REGION} --format json | jq -r .address`
+$ export WORKER_NAT_IP=`gcloud compute addresses describe ${INFRA_ID}-worker-nat-ip --region ${REGION} --format json | jq -r .address`
+----
+
+. Create a `03_iam.yaml` resource definition file:
++
+----
+$ cat <<EOF >03_iam.yaml
+imports:
+- path: 03_iam.py
+resources:
+- name: cluster-iam
+  type: 03_iam.py
+  properties:
+    infra_id: '${INFRA_ID}' <1>
+EOF
+----
+<1> `infra_id` is the `INFRA_ID` infrastructure name from the extraction step.
+
+. Create the deployment by using the `gcloud` CLI:
++
+----
+$ gcloud deployment-manager deployments create ${INFRA_ID}-iam --config 03_iam.yaml
+----
+
+. Export the variable for the master service account:
++
+----
+$ export MASTER_SA=`gcloud iam service-accounts list | grep "^${INFRA_ID}-master-node " | awk '{print $2}'`
+----
+
+. Export the variable for the master service account:
++
+----
+$ export WORKER_SA=`gcloud iam service-accounts list | grep "^${INFRA_ID}-worker-node " | awk '{print $2}'`
+----
+
+. Assign the permissions that the installation program requires to the service accounts for the subnets that host the control plane and compute subnets:
+
+.. Grant the `networkViewer` role of the project that hosts your shared VPC to the master service account:
++
+----
+$ gcloud --account=${HOST_PROJECT_ACCOUNT} --project=${HOST_PROJECT} projects add-iam-policy-binding ${HOST_PROJECT} --member "serviceAccount:${MASTER_SA}" --role "roles/compute.networkViewer"
+----
+
+.. Grant the `networkUser` role to the master service account for the control plane subnet:
++
+----
+$ gcloud --account=${HOST_PROJECT_ACCOUNT} --project=${HOST_PROJECT} compute networks subnets add-iam-policy-binding "${HOST_PROJECT_CONTROL_SUBNET}" --member "serviceAccount:${MASTER_SA}" --role "roles/compute.networkUser" --region ${REGION}
+----
+
+.. Grant the `networkUser` role to the worker service account for the control plane subnet:
++
+----
+$ gcloud --account=${HOST_PROJECT_ACCOUNT} --project=${HOST_PROJECT} compute networks subnets add-iam-policy-binding "${HOST_PROJECT_CONTROL_SUBNET}" --member "serviceAccount:${WORKER_SA}" --role "roles/compute.networkUser" --region ${REGION}
+----
+
+.. Grant the `networkUser` role to the master service account for the compute subnet:
++
+----
+$ gcloud --account=${HOST_PROJECT_ACCOUNT} --project=${HOST_PROJECT} compute networks subnets add-iam-policy-binding "${HOST_PROJECT_COMPUTE_SUBNET}" --member "serviceAccount:${MASTER_SA}" --role "roles/compute.networkUser" --region ${REGION}
+----
+
+.. Grant the `networkUser` role to the worker service account for the compute subnet:
++
+----
+$ gcloud --account=${HOST_PROJECT_ACCOUNT} --project=${HOST_PROJECT} compute networks subnets add-iam-policy-binding "${HOST_PROJECT_COMPUTE_SUBNET}" --member "serviceAccount:${WORKER_SA}" --role "roles/compute.networkUser" --region ${REGION}
```

Comment:
HOST_PROJECT_COMPUTE_SUBNET has not been defined yet. Please keep the name convention consistent with CONTROL_SUBNET.